extensions/eclipse-che-theia-user-preferences/src/node/che-theia-preferences-synchronizer.ts

```diff
@@ -0,0 +1,86 @@
+/********************************************************************************
+ * Copyright (C) 2018-2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+/// <reference types="@theia/core/src/typings/nsfw/index"/>
+
+import { injectable, inject } from "inversify";
+import { readFileSync, writeFileSync } from 'fs';
+import { homedir } from 'os';
+import { sep as PATH_SEPARATOR } from 'path';
+import * as nsfw from 'vscode-nsfw';
+import { CheApiService, Preferences } from "@eclipse-che/theia-plugin-ext/lib/common/che-protocol";
+
+export const THEIA_PREFERENCES_KEY = 'theia-user-preferences';
+export const THEIA_USER_PREFERENCES_PATH = homedir() + PATH_SEPARATOR + '.theia' + PATH_SEPARATOR + 'settings.json';
```

Comment:
FYI : path.resolve()
https://nodejs.org/api/path.html#path_path_resolve_paths
```
path.resolve('wwwroot', 'static_files/png/', '../gif/image.gif');
// If the current working directory is /home/myself/node,
// this returns '/home/myself/node/wwwroot/static_files/gif/image.gif'
```

Reply:
Yes, I had knew it, but do not know why didn't use... will fix. Thanks.